src/main/java/ir/sahab/kafkaconsumer/OffsetTracker.java

```diff
@@ -183,7 +203,7 @@ OptionalLong ack(long offset) {
                         + "offset: {}", offset);
                 return OptionalLong.empty();
             }
-            int pageOffset = (int) (offset % pageSize);
+            int pageOffset = offsetToMargin(offset);
```

Comment:
```suggestion
            int pageOffset = offsetToPageOffset(offset);
```

We have used the word 'margin' for something else. Margins means the head gap of the page.

Reply:
Done